src/App.tsx

```diff
@@ -15,71 +16,77 @@ export const goods = [
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const itemSelected = (item: string) => {
+    setSelectedItam(item);
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      {selectedItem === ''
+        ? (
+          <h1 className="title">No goods selected</h1>
+        )
+        : (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedItem} is selected`}
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+              className="delete ml-3"
+              onClick={() => itemSelected('')}
+            />
+          </h1>
+        )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              key={good}
+              data-cy="Good"
+              className={cn({
+                'has-background-success-light': selectedItem === good,
+              })}
             >
-              +
-            </button>
-          </td>
+              <td>
+                {selectedItem === good
+                  ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={() => itemSelected('')}
+                    >
+                      -
+                    </button>
+                  )
+                  : (
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => itemSelected(good)}
+                    >
+                      +
+                    </button>
+                  )}
```

Comment:
in a spare time think how you could DRY your code
i'd suggest a tiny function that would return the needed button depending on the needed type 
```
const renderButton = (type: 'add' | 'remove') => ...
```